academic-requirements/src/components/DropTarget.tsx

```diff
@@ -259,42 +270,35 @@ export const Container: FC<ContainerProps> = memo(function Container({
           )
         ) {
           // prereqCheck returned true, so add the course to the semester
+          course.dragSource = "Semester " + index;
           setSemesters(
             update(semesters, {
               [index]: {
                 lastDroppedItem: {
                   $set: item,
                 },
                 courses: {
-                  $push: itemArr,
+                  $push: [course],
                 },
               },
             })
           );
-          handleRemoveItem(course);
+          //handleRemoveItem(course);
         } else {
           // fails to satisfy prerequisites
+          //shows error message
           setVisibility(true);
         }
       }
       // Course was not found in the courses list, which means it currently occupies a semester
       else {
-        let course = courses.find((item) => item.name === name);
-
         //Find the course and its current residing index in the semesters list
-        let movedFromIndex = -1;
-        semesters.forEach((x, i) => {
-          x.courses.forEach((y) => {
-            if (y.name === item.name) {
-              course = y;
-              movedFromIndex = i;
-            }
-          });
-        });
-
+        console.log(movedFromIndex);
         let preReqsSatisfied = true;
-
+        console.log(course);
+        console.log(movedFromIndex > -1);
         if (course && movedFromIndex > -1) {
+          console.log("Made it");
```

Comment:
There's a few console.log statements that seemed like they were for testing, but could probably be deleted now. I think the console.log(course) should stay though.

Reply:
Yup, meant to do that.